frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/initialize-vm.ts

```diff
@@ -46,15 +46,16 @@ const initializeStorage = (params: CreateVMParams, vm: VMWrapper) => {
         volumeWrapper.appendTypeData({ name: dataVolumeWrapper.getName() });
       }
     }
-
     if (volumeWrapper.getCloudInitNoCloud()) {
-      const cloudConfigHeader = volumeWrapper
+      const cloudInitNoCloudUserData = volumeWrapper
         .getCloudInitNoCloud()
-        ?.userData?.includes('#cloud-config');
-      !cloudConfigHeader &&
-        volumeWrapper.setCloudInitNoCloud({
-          userData: ['#cloud-config', volumeWrapper.getCloudInitNoCloud().userData].join('\n'),
-        });
+        ?.userData?.replaceAll("'", ''); // remove extra single quotation marks if needed
```

Comment:
This will replace the -'- sign if it exists in the password itself (it's an eligible character in Linux passwords for instance), best will be to remove the first and last character to avoid that

Reply:
@hstastna I would look here

https://github.com/openshift/console/blob/master/frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/default-template.ts#L67

Reply:
Thank you, folks. Yes, I know, `replaceAll` is not a good idea, this is why I moved PR to WIP, when I realized that. Yes, I already checked that place in the code. Thanks! The password seems to be generated correctly, the problem is different... :)

Reply:
i have some issues with that regexp
a. it's hard to read
b. it only catch cases where the password ends with endline `\n` 
    will miss `"password: 'hello' "`
c. it will remove a `'` that is part of the password if it's the last char:
    if will remove the `'` at the end of   `"password: hello'\n"` although it's part of the password.
d. it will remove the last `'` from fields that are not `password:` 
    if will remove the `'` at the end of   `"name: 'hello'\n"` although it's part of the password.
    
I suggest using this regexp:
```js
.replace(/password:\s'(.*)'\s/g, 'password: $1'))
```
I think it's clearer and does not have the above issues

Reply:
/hold cancel

Reply:
/hold

sorry, I made a mistake in my suggested regexp
need to be:
`.replace(/(password:\s)'(.*)'(\s)/g, '$1$2$3')`

in my previous suggestion i forgot the last white space :facepalm: